src/Entity/CrudQueue.php

```diff
@@ -223,6 +258,40 @@ public function sendItem() {
       }
       $created = TRUE;
     }
+    catch (\Aws\Sqs\Exception\SqsException $e) {
+      // For SQS Exceptions, we can gather additional debugging data.
+      $error['response_message'] = format_string(
+        '@message Error type: @type.',
+        [
+          '@message' => $e->getMessage(),
+          '@type' => $e->getAwsErrorType(),
+        ]
+      );
+      $error['response_code'] = $e->getAwsErrorCode();
+      $error['response_raw'] = $e->getResponse();
+    }
+    catch (\Exception $e) {
+      $error['response_message'] = $e->getMessage();
+      $error['response_code'] = $e->getCode();
+    }
+
+    // Pass original keys to notification hook. This hook could allow, for
+    // example, various kinds of reporting.
+    $item_info = [
+      'type' => $this->type,
+      'entity' => $this->entity,
+      'op' => $this->op,
+      'result' => $result,
+      'data' => $data,
+      'message' => [
+        'attributes' => $this->getMessageAttributes(),
+        'id' => $this->getMessageId(),
+      ],
+    ];
+    if (isset($error)) {
+      $debug_info['error'] = $error;
+    }
+    module_invoke_all('aws_sqs_entity_send_item', $item_info);
```

Comment:
I also like a more robust `$item_info`. The prior params were simple, but I can see how having more SQS message item data could be useful.